tests/next_tests/integration_tests/multi_feature_tests/iterator_tests/test_hdiff.py

```diff
@@ -74,7 +74,10 @@ def hdiff(inp, coeff, out, x, y):
 def test_hdiff(hdiff_reference, program_processor, lift_mode):
     program_processor, validate = program_processor
     if program_processor == run_gtfn or program_processor == run_gtfn_imperative:
-        pytest.xfail("origin not yet supported in gtfn")
+        from gt4py.next.iterator import transforms
+
+        if lift_mode != transforms.LiftMode.FORCE_INLINE:
+            pytest.xfail("origin not yet supported in gtfn")
```

Comment:
Can you fix the message? Do you understand what's not working?

Reply:
I don't really know, it crashes when freeing the temporaries, either bad logic or memory corruption. Probably also unrelated to this PR because it works when everything is inlined and the inputs and outputs are the same.